core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -1179,12 +1179,12 @@ public CommitId flush(boolean force, boolean waitIfOngoing) throws EngineExcepti
                     try {
                         translog.prepareCommit();
                         logger.trace("starting commit for flush; commitTranslog=true");
-                        commitIndexWriter(indexWriter, translog, null);
+                        final long committedGeneration = commitIndexWriter(indexWriter, translog, null);
```

Comment:
One naming thing that gets to me is `prepareCommit` and `commit` in `Translog`. I don't think `commit` commits anything. It trims segments based on a commit in permanent storage. I wonder if it'd make the naming easier to stop using the word "commit" for the thing that the Translog is doing. Then any time you see the word "commit" it is about a lucene commit. If you renamed `Translog#commit` into `Translog#updatePersistedGeneration` or something then it'd be more clear that it is reacting to  a commit in Lucene. Then maybe the other variable names just fall out of that?